### ISSUE TITLE
UserUUIDHelper.getUUIDFromXUID seems to be skipping a byte

### ISSUE THREAD

**Issue by rtsai** [Luluno01/BungeeGuard#9]
The following code seems to be skipping the 6th byte:

https://github.com/Luluno01/BungeeSafeguard/blob/36d8b27b7fdfa3919836289282434adec2a1609c/src/main/kotlin/vip/untitled/bungeesafeguard/helpers/UserUUIDHelper.kt#L70-L86

There seems to be missing something like:

```kotlin
${xuid.ushr(40).and(0xff).toString(16).padStart(2, '0')}
```

immediately preceding `xuid.ushr(32)`. The XUIDs I am seeing are reported by Geyser and looked-up by `xbl-web-api` to have a byte `01` in that position, but BungeeSafeguard always yields `00` in that same position:

```
00000000-0000-0000-00AA-[01]BBCCDDEEFF    # Geyser and xbl-web-api
00000000-0000-0000-00AA-[00]BBCCDDEEFF    # BungeeSafeguard
```

Otherwise this plugin works great for Java clients, and I am able to workaround for Bedrock clients by using `lazy-add`. I don't have a Kotlin development setup so I cannot build for myself to test.

A possible simplification of this code might be to mask off multiple bytes at a time, rather than a byte-at-a-time (not sure if this is valid Kotlin syntax, just kind of guessing):

```kotlin
"00000000-0000-0000-${xuid.ushr(48).and(0xffff).toString(16).padStart(4, '0')
}-${xuid.and(0xffffffffffff).toString(16).padStart(12, '0')}"
```

**Comment by Luluno01** [comment]
Thanks for your detailed report. I will look into it asap.

**Comment by Luluno01** [comment]
Fixed in [v2.6.1](https://github.com/Luluno01/BungeeSafeguard/releases/tag/v2.6.1).

BTW, this plugin is going under massive refactoring to prepare for possible future changes (which are unseen by now, but just in case).
